### ISSUE TITLE
[Question] Starting Treemacs on creation of new frame and unfocusing it

### ISSUE THREAD

**Issue by Nekosis** [Alexander-Miller/treemacs#1060]
Hi,

I have this as part of my [Emacs configuration](https://github.com/Nekosis/nyamacs#treemacs):

```elisp
(use-package
 treemacs
 :after treemacs-all-the-icons
 :config
 (setq treemacs-width 42)
 (treemacs-load-theme "all-the-icons")
 :hook
 (server-after-make-frame . treemacs))
(use-package treemacs-all-the-icons :after all-the-icons)
```

As you can see, I have a hook to start Treemacs after creation of a new frame.

However, Treemacs is focused when this happens. I would like to have Treemacs unfocused, and instead have the file requested to open (or the dashboard if no file is provided) focused instead.

How can I do this?

**Comment by Nekosis** [comment]
Figured it out. See Nekosis/nyamacs@8dea834d03a93fa9d7f700c09b509b2b31e18f74

**Comment by Alexander-Miller** [comment]
If you want to do a thing, and stay in the same window afterwards, you can wrap you code in `save-selected-window`.

**Comment by Nekosis** [comment]
Thank you for your help @Alexander-Miller, I've added `save-selected-window` in commit Nekosis/nyamacs@12eb7ac. One more question, if you don't mind; when I start `emacsclient` with this config, I get this strange red highlight line. Do you know what might be causing this?

https://github.com/Alexander-Miller/treemacs/assets/37462865/6b39927d-dc22-46fb-883e-1a36d4004e0f

**Comment by Alexander-Miller** [comment]
Could be that  treemacs-fringe-indicator-modeis getting confused. 

Looking at your commit the code might also be an issue because the save-window doesn't do anything. What you want to do is create a hook that looks something like this:

```Elisp
(save-selected-window
  (treemacs)
  (hl-line-highlight))
```

You shouldn't need to expand a node yourself if `treemacs-expand-after-init` is t, which you can also let-bind. Only the hl-line call is needed to fix its highlight.

**Comment by Nekosis** [comment]
Ah, I see what you mean now. I didn't actually need the hl-line-highlight; the `treemacs-post-buffer-init` hook seemed to be what was causing the problems.

```elisp

Finally fixed as of Nekosis/nyamacs@b293250. Thank you so much for your help!
